src/python/pants/backend/python/subsystems/python_tool_base.py

```diff
@@ -34,11 +36,15 @@ class PythonToolRequirementsBase(Subsystem):
     # Subclasses do not need to override.
     default_extra_requirements: ClassVar[Sequence[str]] = []
 
+    # Subclasses may set to override the value computed from default_version and
+    # default_extra_requirements.
+    # TODO: Once we get rid of those options, subclasses must set this.
+    default_package_names: Sequence[str] = []
```

Comment:
I think package suffers from overloading in this space. This is not a package as in module/package - it's a distribution package (as you word it elsewhere in this PR). The distribution package - a .whl or .tar.gz/.zip - is the wrong layer of abstraction though. I think s/package/project/ correctly names this. Now correct vs what a User will understand are two different things that don't always coincide. So, with all that - as you see fit.

Reply:
My vote leans towards correctness with docs/help to aid users when/where needed--hopefully leads to fewer surprises overall.

Reply:
Yeah I wrote and rewrote this name six times before I landed on "package", as short for "distribution package", and I'm still very unsure that is the right name. I can change to `project`, you're right that this is the most technically correct. And since I'm not sure what users will find comprehensible, we may as well fall back on technical correctness.